crates/tree-sitter-perl-c/c-src/bsearch.h

```
@@ -0,0 +1,90 @@
+ * is odd, moving left simply involves halving lim: e.g., when lim
+ * is 5 we look at item 2, so we change lim to 2 so that we will
+ * look at items 0 & 1.  If lim is even, the same applies.  If lim
+ * is odd, moving right again involes halving lim, this time moving
+ * the base up one item past p: e.g., when lim is 5 we change base
+ * to item 3 and make lim 2 so that we will look at items 3 and 4.
+ * If lim is even, however, we have to shrink it by one before
+ * halving: e.g., when lim is 4, we still looked at item 2, so we
+ * have to make lim 3, then halve, obtaining 1, so that we will only
+ * look at item 3.
+ */
+void *bsearch(const void *key, const void *base0, size_t nmemb, size_t size,
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'bsearch' of similar type ('const void *') are easily swapped by mistake

(bugprone-easily-swappable-parameters)